client/deployments/client.go

```diff
@@ -49,15 +88,92 @@ func NewClient(url string, skipVerify bool) *Client {
 
 	return &Client{
 		url:               url,
-		artifactUploadUrl: JoinURL(url, artifactUploadUrl),
+		artifactUploadURL: JoinURL(url, artifactUploadURL),
+		artifactsListURL:  JoinURL(url, artifactsListURL),
 		client: &http.Client{
 			Transport: tr,
 		},
 	}
 }
 
-func (c *Client) UploadArtifact(description, artifactPath, tokenPath string, noProgress bool) error {
+func (c *Client) ListArtifacts(tokenPath string, detailLevel int) error {
+	token, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return errors.Wrap(err, "Please Login first")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, c.artifactsListURL, nil)
+	if err != nil {
+		return errors.Wrap(err, "Cannot create request")
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, _ := httputil.DumpRequest(req, false)
+	log.Verbf("sending request: \n%v", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "Get /deployments/artifacts request failed")
+	}
+	defer rsp.Body.Close()
+
+	body, err := ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return err
+	}
+
+	var list artifactsList
+	err = json.Unmarshal(body, &list.artifacts)
+	if err != nil {
+		return err
+	}
+	for _, v := range list.artifacts {
+		listArtifact(v, detailLevel)
+	}
+
+	return nil
+}
+
+func listArtifact(a artifactData, detailLevel int) {
```

Comment:
This could potentially be a nice use-case for Go's text template package: https://golang.org/pkg/text%2Ftemplate/

But this is no requirement, this is just fine. But if you're feeling adventurous :)

Reply:
thx for the hint. Will look into it, but leaving this for now the way it is.